OpenRA.Game/Selection.cs

```diff
@@ -33,40 +33,65 @@ public bool Contains(Actor a)
-				}
-			}
-			else
-			{
-				if (isCombine)
-					actors.UnionWith(newSelection);
-				else
-				{
-					actors.Clear();
-					actors.UnionWith(newSelection);
-				}
-			}
-
-			var voicedUnit = actors.FirstOrDefault(a => a.Owner == world.LocalPlayer && a.IsInWorld && a.HasVoices());
-			if (voicedUnit != null)
-				Sound.PlayVoice("Select", voicedUnit, voicedUnit.Owner.Country.Race);
-
-			foreach (var a in newSelection)
-				foreach (var sel in a.TraitsImplementing<INotifySelected>())
-					sel.Selected(a);
-			foreach (var ns in world.WorldActor.TraitsImplementing<INotifySelection>())
-				ns.SelectionChanged();
-		}
+        public void Combine(World world, IEnumerable<Actor> newSelection, bool isCombine, bool isClick)
+        {
+            var classicmouse = Game.Settings.Game.UseClassicMouseStyle;
+            if (isClick)
+            {
+                var adjNewSelection = newSelection.Take(1);	/* TODO: select BEST, not FIRST */
+                if (isCombine)
+                    actors.SymmetricExceptWith(adjNewSelection);
+                else
+                {
+                    if (classicmouse == true)
+                    {
+                        if (newSelection.Count() == 0)
+                        {
+                            actors.UnionWith(adjNewSelection);
+                        }
+                            else
+                                actors.Clear();
+                                actors.UnionWith(newSelection);
+                    }
+                    else
+
+                        actors.Clear();
+                        actors.UnionWith(newSelection);
+                }
+            }
+            else
+            {
+                if (isCombine)
+                    actors.UnionWith(newSelection);
+                else
+                {
+                    if (classicmouse == true)
+                    {
+                        if (newSelection.Count() == 0)
+                            actors.UnionWith(newSelection);
+                        else
+                            actors.Clear();
+                            actors.UnionWith(newSelection);
+                    }
+                    else
+                        actors.Clear();
+                        actors.UnionWith(newSelection);
+               }
+            }
+
+            var voicedUnit = actors.FirstOrDefault(a => a.Owner == world.LocalPlayer && a.IsInWorld && a.HasVoices());
+            if (newSelection.Count() != 0)
+            {
+            if (voicedUnit != null)
+                Sound.PlayVoice("Select", voicedUnit, voicedUnit.Owner.Country.Race);
+                                           }
+
+            foreach (var a in newSelection)
+                foreach (var sel in a.TraitsImplementing<INotifySelected>())
+                    sel.Selected(a);
+            foreach (var ns in world.WorldActor.TraitsImplementing<INotifySelection>())
+                ns.SelectionChanged();
+        }
```

Comment:
Can you replace the space indention with tabs please. We provide an @EditorConfig that may be of help to you.